src/ResponseTagger.php

```diff
@@ -36,10 +50,31 @@ class ResponseTagger
      * Constructor
      *
      * @param TagsInterface $client
+     * @param array         $options supported options:
+     *                               - strict (bool) Default: false. If set to true, throws exception when adding empty tags.
      */
-    public function __construct(TagsInterface $client)
+    public function __construct(TagsInterface $client, array $options = array())
     {
         $this->client = $client;
+
+        if (!isset(self::$resolver)) {
+            self::$resolver = new OptionsResolver();
+            $this->configureOptions(self::$resolver);
+        }
+
+        $this->options = self::$resolver->resolve($options);
+    }
+
+    /**
+     * Configure options
+     *
+     * @param OptionsResolver $resolver
+     */
+    public function configureOptions(OptionsResolver $resolver)
```

Comment:
why public and not private? and why a method? i would inline this in the constructor. and not use a static variable either, just a local variable. i don't expect there to be more than one response tagger.